apps/src/codebridge/FileBrowser/index.tsx

```diff
@@ -434,6 +435,15 @@ export const FileBrowser = React.memo(() => {
   );
   const isStartMode = getAppOptionsEditBlocks() === START_SOURCES;
 
+  const checkForDuplicateFilename = usePartialApply(
+    globalCheckForDuplicateFilename,
+    {
+      isStartMode,
+      validationFile,
+      projectFiles: project.files,
+    } satisfies PAFunctionArgs<typeof globalCheckForDuplicateFilename>
+  );
+
```

Comment:
This way we don't need to keep handing the extra flags and the list of project files into the `checkForDuplicateFilename` function.

Admittedly, we then _do_ need to hand in this function to the various `*Prompt` dependency arrays, but that's what I'm going to be cleaning up across the next several commits.